pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -222,7 +227,7 @@ func (configgen *ConfigGeneratorImpl) buildSidecarListeners(env *model.Environme
 		}
 
 		var transparent *google_protobuf.BoolValue
-		if mode := node.Metadata["INTERCEPTION_MODE"]; mode == "TPROXY" {
+		if mode := node.Metadata[pilot.InterceptionMode]; mode == pilot.InterceptionModeTproxy {
 			transparent = proto.BoolTrue
 		}
 
```

Comment:
For 'stateful sets' - which can't be supported otherwise yet ( we can add SOCKS support in envoy at some point - it's also almost universally supported )